addOns/grpc/src/main/java/org/zaproxy/addon/grpc/internal/VariantGrpc.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2024 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.grpc.internal;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.List;
+import java.util.Objects;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.core.scanner.NameValuePair;
+import org.parosproxy.paros.core.scanner.Variant;
+import org.parosproxy.paros.network.HttpMessage;
+
+public class VariantGrpc implements Variant {
+    private static final Logger LOGGER = LogManager.getLogger(VariantGrpc.class);
+    private final List<NameValuePair> params = new ArrayList<>();
+    private final ProtoBufMessageEncoder protoBufMessageEncoder = new ProtoBufMessageEncoder();
+    private final ProtoBufMessageDecoder protoBufMessageDecoder = new ProtoBufMessageDecoder();
+
+    private String requestDecodedBody = null;
+
+    @Override
+    public void setMessage(HttpMessage msg) {
+        if (isValidContentType(msg)) {
+            try {
+                byte[] body = Base64.getDecoder().decode(msg.getRequestBody().getBytes());
+                byte[] payload = DecoderUtils.extractPayload(body);
+                protoBufMessageDecoder.decode(payload);
+                parseContent(protoBufMessageDecoder.getDecodedToList(), "");
+                requestDecodedBody = protoBufMessageDecoder.getDecodedOutput();
+            } catch (Exception e) {
+                LOGGER.error("Parsing message body failed: {}", e.getMessage());
+            }
+        }
+    }
+
+    private void parseContent(List<String> decodedList, String commonPrefixForNestedMessage)
+            throws InvalidProtobufFormatException {
+        for (String pair : decodedList) {
+            String[] nameValuePair = pair.split("::", 2);
+            if (commonPrefixForNestedMessage.isEmpty()) {
+                params.add(
+                        new NameValuePair(
+                                NameValuePair.TYPE_GRPC_WEB_TEXT,
```

Comment:
You cannot use this constant until after a new full release. You can use a local const for now, and add a `TODO` to update it when theres a new ZAP release